src/crab/ebpf_domain.cpp

```diff
@@ -1334,6 +1375,42 @@ void ebpf_domain_t::assign_valid_ptr(const Reg& dst_reg, bool maybe_null) {
     m_inv += reg.value <= PTR_MAX;
 }
 
+// If nothing is known of the stack_numeric_size,
+// try to recompute the stack_numeric_size.
+void ebpf_domain_t::recompute_stack_numeric_size(NumAbsDomain& inv, variable_t type_variable) {
+    variable_t stack_numeric_size_variable = variable_t::kind_var(data_kind_t::stack_numeric_sizes, type_variable);
+
+    if (!inv.eval_interval(stack_numeric_size_variable).is_top())
+        return;
```

Comment:
Tried your suggestion and it results in a bunch of other invariants being added unrelated to registers in the yaml test case.  E.g.:
```
Filters: [yaml]
test case: stack assign immediate
unseen inv: []
unexpected inv: [
    r1.stack_numeric_size=[0, +oo],
    r10.stack_numeric_size=[0, +oo],
    r2.stack_numeric_size=[0, +oo]]
```

I haven't tried to track this down, as I hope it isn't a blocker, so maybe it can be done in a future PR.

Reply:
Not a blocker. These are the invariants I've mentioned (they are just TOP, really).